man/checkers/cstyleCast.md

```diff
@@ -0,0 +1,76 @@
+
+# cstyleCast
+
+**Message**: C-style pointer casting<br/>
+**Category**: Type Safety<br/>
+**Severity**: Style<br/>
+**Language**: C++, not applicable for C code
+
+## Motivation
+
+C style casts can be dangerous in many ways:
+ * there can be unintentional loss of precision
+ * there can be unintentional loss of sign
+ * there can be unintentional loss of constness
+ * there can be invalid type conversions
+
+## Philosophy
+
+This checker warns about old style C casts that looks dangerous.
+
+This checker is not about readability. It is about safety.
+
+We only want to warn if there are possible safety issues and a C++ cast can provide better safety.
+
+## Other tools / checkers
+
+Related checkers:
+ * Complements: Misra C 11.8 is violated if there is "loss of constness"
+ * More noisy: Misra C++ 8.2.2, cover all C style casts.
+ * More noisy: C style cast compiler warnings (i.e. clang reports Wold-style-cast)
+ * Complements: Loss of constness compiler warnings.
+
+## Example
```

Comment:
Having examples in the documentation obviously helps but it provides incomplete code which might be misleading.

We also have the `examples` which would provide way more details and an actual sample file to analyze. Maybe we should merge that with the documentation.

Reply:
> Having examples in the documentation obviously helps but it provides incomplete code which might be misleading.

I made the code complete

> We also have the examples which would provide way more details and an actual sample file to analyze. Maybe we should merge that with the documentation.

I do like to have the code directly in the documentation. If we merge it somehow I am not against it basically. But personally I would drop the examples folder and just provide the documentation.

Reply:
Having the `examples` separate helps making sure that the code is valid and is always detected. Also people use this to test their setups. We could inject the examples via a script.

Reply:
> We could inject the examples via a script.

ok

Reply:
> Having the `examples` separate helps making sure that the code is valid and is always detected. Also people use this to test their setups. We could inject the examples via a script.

This is not just personal preference (i.e. avoid duplicated/manual work) but based on actual feedback I have seen with Cppcheck and clang-tidy (they have/had several broken code examples in documentation) - mostly from first-time users.

I am not blocking this PR but it is something we should consider moving forward.

We could also leverage the effort of documenting/adding examples for the checks to fix the holes in the unit testing - see https://trac.cppcheck.net/ticket/12232.